contexts/Aladin.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+import {
+  createContext,
+  FunctionComponent,
+  MutableRefObject,
+  PropsWithChildren,
+  RefCallback,
+  useCallback,
+  useContext,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  Aladin,
+  AladinInstance,
+  AladinOptions,
+  ImageHiPSOptions,
+} from "@/types/aladin";
+import defaultOptions, {
+  defaultHiPSOptions,
+} from "@/fixtures/defaultAladinOptions";
+import { useOnClickOutside } from "@/hooks/listeners";
+
+interface AladinProps {
+  fovRange?: Array<number>;
+  options?: AladinOptions;
+  hipsConfig: {
+    id: string;
+    options: ImageHiPSOptions;
+  };
+}
+
+interface AladinContextValue {
+  aladin: AladinInstance | null;
+  A: Aladin | null;
+  setRef: RefCallback<HTMLElement>;
+  ref: MutableRefObject<HTMLElement | null>;
+  hasFocus: boolean;
+}
+
+const AladinContext = createContext<AladinContextValue | null>(null);
+
+export const AladinProvider: FunctionComponent<
```

Comment:
this provider and hook are the most critical part of the v3 upgrade, this provider attempts to load Aladin into React as a side effect and make it available to other React components